core/src/main/scala/org/apache/spark/executor/ShuffleReadMetrics.scala

```diff
@@ -146,6 +268,16 @@ private[spark] class TempShuffleReadMetrics extends ShuffleReadMetricsReporter {
   override def incLocalBytesRead(v: Long): Unit = _localBytesRead += v
   override def incFetchWaitTime(v: Long): Unit = _fetchWaitTime += v
   override def incRecordsRead(v: Long): Unit = _recordsRead += v
+  override def incCorruptMergedBlockChunks(v: Long): Unit = _corruptMergedBlockChunks += v
+  override def incFallbackCount(v: Long): Unit = _fallbackCount += v
+  override def incRemoteMergedBlocksFetched(v: Long): Unit = _remoteMergedBlocksFetched += v
+  override def incLocalMergedBlocksFetched(v: Long): Unit = _localMergedBlocksFetched += v
+  override def incRemoteMergedChunksFetched(v: Long): Unit = _remoteMergedChunksFetched += v
+  override def incLocalMergedChunksFetched(v: Long): Unit = _localMergedChunksFetched += v
+  override def incRemoteMergedBlocksBytesRead(v: Long): Unit = _remoteMergedBlocksBytesRead += v
+  override def incLocalMergedBlocksBytesRead(v: Long): Unit = _localMergedBlocksBytesRead += v
+  override def incRemoteReqsDuration(v: Long): Unit = _remoteReqsDuration += v
+  override def incRemoteMergedReqsDuration(v: Long): Unit = _remoteMergedReqsDuration += v
```

Comment:
I am able to find the calling for these two methods in SHS which can build the metrics up from JSON, but the caller in the ShuffleBlockFetcherIterator is actually missing in this patch. For the caller part in Executor, which measures the remote request duration, needs more refactoring in the Iterator. Probably we can carve these two metrics out in this PR. If needed, we can create another PR which tracks the remote request duration for unmerged and merged shuffle. @mridulm @otterc Thoughts?

Reply:
I called out the same [above](https://github.com/apache/spark/pull/36165#discussion_r978229580)
I am fine with pulling those two metrics out and introducing them in a different PR